### PR TITLE
feat: check for solc when running evm ops

### PR DESCRIPTION
**Pull request by gagichce**
Feature: Adds check for `solc` in env before running evm operations

Ran into this issue myself and thought I'd make it easier for everyone!

I wanted to keep the check agnostic where possible as there's many ways `solc` can be installed. However I'm happy to add another check for the case mentioned in the [original issue](https://github.com/zkonduit/ezkl/issues/346
) where `solc-select` is installed but `SOLC_VERSION` is not set.

Checks a few possible scenarios:
- `solc` not installed in ENV at all
- `solc` corrupt / otherwise not working
- `solc` not configured when using `solc-select`

Might be worth checking which version is installed, but for a future issue ;P

### CODE REVIEW

**Comment by alexander-camuto** [comment]
Hey @gagichce, thanks so much for this :)

File: src/execute.rs [hunk 1]
```diff
@@ -66,6 +72,35 @@ use std::sync::atomic::{AtomicBool, AtomicI64, Ordering};
 #[cfg(not(target_arch = "wasm32"))]
 use std::time::Duration;
 use thiserror::Error;
+
+#[cfg(not(target_arch = "wasm32"))]
+static _SOLC_REQUIREMENT: OnceLock<bool> = OnceLock::new();
+#[cfg(not(target_arch = "wasm32"))]
+fn check_solc_requirement() {
+    info!("checking solc installation..");
+    _SOLC_REQUIREMENT.get_or_init(|| {
```
Comment (gagichce) [inline src/execute.rs:81]: Added `solc` check into `OnceLock` for a threadsafe way of running this only once per exec.

@alexander-camuto

**Comment by JSeam2** [comment]
With regards to solc versions there might be some conflicts with pre-Shanghai EVM and post-Shanghai EVM. The issue is noted in #349. I think this PR should be fine for now.

**Comment by alexander-camuto** [comment]
Hey @gagichce last lingering issue. Seems like the solc install is being checked at times when it shouldn't be. For eg. 
image

**Review by ethan-crypto — COMMENTED** [review]
Overall I really appreciate the PR! A lot of folks will find this helpful. 
Also can you add the solc checks to the deploy_evm and deploy_da_evm commands? Essentially any function that transacts with the evm or compiles a solidity file should have the check, and all others shouldn't.

File: src/execute.rs [hunk 2]
```diff
@@ -1047,6 +1086,7 @@ pub(crate) async fn prove(
 ) -> Result<Snark<Fr, G1Affine>, Box<dyn Error>> {
     let data = GraphWitness::from_path(data_path)?;
     use crate::pfsys::load_pk;
+    check_solc_requirement();
```
[the remark this answers is not in the document]
Comment (gagichce) [reply src/execute.rs]: Thanks for the feedback! I've made the changes.

**Comment by gagichce** [comment]
Thanks for the review all! I've updated the PR .. let me know if there's anything else you'd like to see.

**Comment by JSeam2** [comment]
Running the test suites once more to double check

**Review by JSeam2 — APPROVED** [review]
LGTM

**Pull request merged**